src/app/GitCommands/Git/Commands.Execution.cs

```diff
@@ -0,0 +1,85 @@
+﻿using GitExtensions.Extensibility;
+using GitExtensions.Extensibility.Git;
+using GitExtUtils;
+
+namespace GitCommands.Git;
+
+public static partial class Commands
+{
+    /// <summary>
+    ///  Get all config settings from git according to the scope.
+    /// </summary>
+    /// <param name="gitExecutable">The <see cref="IGitModule.GitExecutable"/> for the affected repo.</param>
+    /// <param name="settingLevel">The scope for the config.</param>
+    /// <returns>The names and values of the settings delimited by '\0'.</returns>
+    public static string GetGitSettings(this IExecutable gitExecutable, GitSettingLevel settingLevel)
+    {
+        GitArgumentBuilder args = new("config")
+            {
+#if UseNewGitConfigSyntax2_46
+                "list",
+#else
+                "--list", // deprecated, but it is too early to force users to update git
+#endif
+                settingLevel switch
+                {
+                    GitSettingLevel.Effective => "",
+                    GitSettingLevel.Local => "--local",
+                    GitSettingLevel.Global => "--global",
+                    GitSettingLevel.SystemWide => "--system",
+                    _ => throw new ArgumentOutOfRangeException(nameof(settingLevel))
+                },
+                "--includes",
+                "--null"
+            };
+        ExecutionResult result = gitExecutable.Execute(args, throwOnErrorExit: false);
+
+        if (result.ExitedSuccessfully)
+        {
+            return result.StandardOutput;
+        }
+
+        if (result.StandardError.StartsWith("fatal: unable to read config file") && result.StandardError.EndsWith(": No such file or directory"))
+        {
+            return "";
+        }
+
+        result.ThrowIfErrorExit("Error getting config values");
+        return "unreachable code";
+    }
+
+    /// <summary>
+    ///  Sets or unsets a git config setting.
+    /// </summary>
+    /// <param name="gitExecutable">The <see cref="IGitModule.GitExecutable"/> for the affected repo.</param>
+    /// <param name="settingLevel">The scope for the config (must not be <see cref="GitSettingLevel.Effective"/>).</param>
+    /// <param name="setting">The name of the setting (may contain dots, e.g. "core.autocrlf").</param>
+    /// <param name="value">The value of the setting or <see langword="null"/> for removing the setting.</param>
+    public static void SetGitSetting(this IExecutable gitExecutable, GitSettingLevel settingLevel, string setting, string? value)
+    {
+        bool isSet = !string.IsNullOrEmpty(value);
+        GitArgumentBuilder args = new("config")
+            {
+#if UseNewGitConfigSyntax2_46
```

Comment:
Is there a particular reason to not use `GitVersion.Current` (i.e. creating a property `GitVersion.Current.SupportNewGitConfigSyntax`) instead?